### PR TITLE
jquery-ajax diagnostic

### PR DESCRIPTION
**Pull request by akadane87**

### CODE REVIEW

File: diagnostic.md [hunk 1]
```diff
@@ -73,14 +104,30 @@ data in JSON format.
 ```
 
 ```sh
-# your answer here
+curl --include --request POST "http://localhost:3000/books/$ID" \
+    --header "Content-Type: application/json" \
+    --data '{
+      "book": {
+        "name": "French Cruller",
```
Comment (RealWeeks) [inline diagnostic.md:111]: tasty book bro

**Comment by akadane87** [comment]
Hahaha thanks bro!

File: diagnostic.md [hunk 2]
```diff
@@ -21,13 +21,22 @@ Write the `curl` request you'd use to retrieve a list of **all donuts** on the
 server.
 
 ```sh
-# your answer here
+curl --include --request "GET" "http://localhost:4741/donuts" \
+  --header "Content-Type: application/json" \
```
Comment (BenGitsCode) [inline diagnostic.md:25]: `curl --include --request GET http://www.example.com/donuts`

File: diagnostic.md [hunk 3]
```diff
@@ -21,13 +21,22 @@ Write the `curl` request you'd use to retrieve a list of **all donuts** on the
...
 
 ```js
-let getDonuts = /* your answer here */;
+let getDonuts = function(id){
+  return $.ajax({
+    url: app.host + '/donuts',
+    method: 'GET',
+  })
+};
 ```
```
Comment (BenGitsCode) [inline diagnostic.md:40]:
``` js
let getDonuts = function () {
  return $.ajax({
    url: 'http://www.example.com/donuts',
    method: 'GET',
  });
};
```

Don't use app.host here, as we shouldnt assume we have that variable estabilished.

File: diagnostic.md [hunk 4]
```diff
@@ -36,13 +45,24 @@ Now, we want to get a **single donut** from the server. Write the `curl` request
...
 
 ```js
-let getDonut = /* your answer here */;
+let getDonuts = function(id){
+  return $.ajax({
+    url: app.host + '/donuts/' + id,
+    method: 'GET',
+  })
+};
 ```
```
Comment (BenGitsCode) [inline diagnostic.md:66]:
``` js
let getDonut = function (donutID) {
  return $.ajax({
    url: 'http://www.example.com/donuts/' + donutID,
    method: 'GET',
  });
};
```

File: diagnostic.md [hunk 5]
```diff
@@ -51,13 +71,24 @@ Write the `curl` request you'd use to delete a single donut, using whatever
...
 ```sh
-# your answer here
+curl --include --request "DELETE" "http://localhost:4741/donuts/$ID" \
+  --header "Content-Type: application/json" \
+  --data '{
+    "donuts": {
+      "name": "Chocolate",
+    }
+  }'
 ```
```
Comment (BenGitsCode) [inline diagnostic.md:81]: `curl --include --request DELETE http://www.example.com/donuts/1`

File: diagnostic.md [hunk 6]
```diff
@@ -36,13 +45,24 @@ Now, we want to get a **single donut** from the server. Write the `curl` request
...
 ```sh
-# your answer here
+curl --include --request "GET" "http://localhost:4741/donuts/ID" \
+  --header "Content-Type: application/json" \
+  --data '{
+    "donuts": {
+      "name": "Chocolate",
+    }
+  }'
 ```
```
Comment (BenGitsCode) [inline diagnostic.md:55]: `curl --include --request GET http://www.example.com/donuts/1`

File: diagnostic.md [hunk 7]
```diff
@@ -51,13 +71,24 @@ Write the `curl` request you'd use to delete a single donut, using whatever
...
 
 ```js
-let deleteDonut = /* your answer here */;
+let deleteDonuts = function(id){
+  return $.ajax({
+    url: app.host + '/donuts/' + id,
+    method: 'DELETE',
+  })
+};
 ```
```
Comment (BenGitsCode) [inline diagnostic.md:92]:
``` js
let deleteDonut = function (donutID) {
  return $.ajax({
    url: 'http://www.example.com/donuts/' + donutID,
    method: 'DELETE',
  });
};
```

File: diagnostic.md (hunk 1 above)
Comment (BenGitsCode) [reply diagnostic.md:111]:
```
curl --include --request "POST" "http://example.com/donuts" \
  --header "Content-Type: application/json" \
  --data $'{
    "donut": {
      "name": "French Cruller",
      "price": "$0.99"
    }
  }'
```

File: diagnostic.md [hunk 8]
```diff
@@ -73,14 +104,30 @@ data in JSON format.
...
+  return $.ajax({
+    url: app.host + '/donuts' + data.donuts.id,
+    method: 'POST',
+    data: {
+      "name": "French Cruller",
+      "price": "$0.99"
+    }
+  })
+};
 ```
```
Comment (BenGitsCode) [inline diagnostic.md:131]:
``` js
let createDonut = function () {
  return $.ajax({
    url: 'http://example.com/donuts',
    method: 'POST',
    data: {
      "donut" {
        "name": "French Cruller",
        "price": "$0.99"
      }
    }
  });
};
```

File: diagnostic.md [hunk 9]
```diff
@@ -100,11 +147,25 @@ object.
...
-# your answer here
+curl --include --request PATCH "http://localhost:3000/books/$ID" \
+    --header "Content-Type: application/json" \
+    --data '{
+      "book": {
+        "name": "an example donut",
+      }
+    }'
+
 ```
```
Comment (BenGitsCode) [inline diagnostic.md:158]: books?

``` sh
curl --include --request "PATCH" "http://example.com/donuts/42" \
  --header "Content-Type: application/json" \
  --data $'{
    "donut": {
      "name": "Krüller",
    }
  }'
```

File: diagnostic.md [hunk 10]
```diff
@@ -100,11 +147,25 @@ object.
...
 
 Write an AJAX request to change the donut on the server using JSON.
 
 ```js
```
Comment (BenGitsCode) [inline diagnostic.md:162]:
``` js
let changeDonut = function (donutID) {
  return $.ajax({
    url: 'http://example.com/donuts/' + donutID,
    method: 'PATCH',
    data: {
      "donut" {
        "name": "Krüller",
      }
    }
  });
};
```

**Review by BenGitsCode — COMMENTED** [review]
4/10

Please review solutions and reach out for 1:1 if they don't make sense to you.

**Pull request closed without merging**